### PR TITLE
fix(flags): move ff overrides to use local browser state

### LINKED ISSUE #7182: Make feature flags overrides logic perform better
## Is your feature request related to a problem?

We query active feature flags every time for a /decide endpoint call as well as for every event ingested. This means it's critical we're able to calculate this in an efficient manner.

Currently, calculating feature flags works something like this:
1. We query the flag + persons (or upcoming, groups) which flags and variants are active. This includes a distinct_id => person lookup
2. After every flag has been queried, we do a distinct_id => person query (in a subquery) query all FeatureFlagOverrides related to the user and load all the flags all over again

Related problem is that this is these two steps are done separately and step 2 is not covered with in-depth tests. Also exported functions from feature flags module are confusing as you have feature_flag.match, `get_active_feature_flags` and `get_overridden_feature_flags` functions which all do subtly different things.

## Describe the solution you'd like

1. Integrate the overrides logic into the main loop
2. Load overrides alongside feature flags, avoid unneeded second distinct_id => person_id lookup
3. Add tests checking the amount of DB queries done when loading feature flags
4. Unify the API for the module in a more clear - have a single external entrypoint
5. Maybe figure out a way to query Persons once instead of per-flag?

## Describe alternatives you've considered

Not doing this.

## Additional context

Related to #7115 and experimentation milestones. We should clean this up.

cc @rcmarron for context - not worth tackling right now due to groups changes occurring within the module.

### PR DESCRIPTION
**Pull request by rcmarron**
## Problem

Feature flag overrides are a little used feature that greatly impact perf because it adds another query to each `/decide` request.

More details: https://github.com/PostHog/posthog/issues/7182

Closes #7182  #10445

## Changes

This `/decide` query is only needed if the user wants their FF overrides to persist across devices + libraries, but we if we don't worry about this, we can store the override locally, and avoid this query.

To make it clear how this works, this PR adds a note to the top of the override component:

image

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

CI passes + verified overriding FFs works as expected.

### CODE REVIEW

**Comment by rcmarron** [comment]
Note, we should add a deprecation notice to the next release notes that mentions this change.

**Review by macobo — COMMENTED** [review]
Didn't dig deep wrt technical details, but this is looking great!

**Review by neilkakkar — APPROVED** [review]
Tested it out, works like a charm.

Backend looks great, frontend looks good, but I'd love a second look at the frontend (maybe @pauldambra since toolbar is your domain xD )

**Comment by neilkakkar** [comment]
(I'll follow up with a `/decide` cleanup, but don't want to put these two things in the same PR)

File: frontend/src/toolbar/flags/FeatureFlags.tsx [hunk 1]
```diff
@@ -3,131 +3,103 @@ import './featureFlags.scss'
...
-                            value_for_user_without_override,
-                            override,
-                            hasVariants,
-                            currentValue,
-                        }) => {
-                            return (
-                                <div className={'feature-flag-row'}>
+            <div className="local-feature-flag-override-note">
+                <Typography.Text>
+                    Note, overriding feature flags below will only impact the experience within this browser.
```
Comment (pauldambra) [inline frontend/src/toolbar/flags/FeatureFlags.tsx:21]: nit... Most people's reading age is surprisingly low

Is it clearer to say "Note, overriding feature flags below will only affect this browser."

**Review by pauldambra — APPROVED** [review]

**Pull request merged**